session_0/item.rb

```diff
@@ -23,23 +23,39 @@ def initialize(name: '', price: 0, quantity: 1, category_id: 4,
   # Returns a boolean value whether than item is discounted i.e. the
   # discount deadline has been crossed or not.
   def discounted?
-    raise NotImplementedError # TODO
+    if (Time.now < @discount_deadline)
+      return true
+    else
+      return false
+    end
   end
 
-  # If the item is discounted, the current price is 
+  # If the item is discounted, the current price is
   # `price * (100 - discount rate) / 100`. Otherwise, it is same as the price.
   #
   # TODO: Implement instance method 'current_price'
-
+  def current_price
+    if discounted?
+      @current_price = @price * (100 - discount_rate) / 100
+    else
+      @current_price = @price
+    end
+  end
   # The stock price of item is defined as product of current price and
   # quantity.
-  # 
+  #
   # This function takes an Array of items as the parameter and returns
   # a hash with the category id as the key and sum of stock
   # prices of all items of that category as the value.
   #
   # Note: If there are no items for category, stock price for category
   # should be zero.
   def self.stock_price_by_category(items)
+    stock_price = Hash.new
+    (1..4).each {|x| stock_price[x] = 0}
```

Comment:
Instead of hardcoding as `1..4`, can you re-write this in terms of `Item::CATEGORIES`? If you add new categories, you will have to edit code in multiple places